lib/controllers/dialogs-controller.js

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import InitDialog from '../views/init-dialog';
+import CloneDialog from '../views/clone-dialog';
+import CredentialDialog from '../views/credential-dialog';
+import OpenIssueishDialog from '../views/open-issueish-dialog';
+import OpenCommitDialog from '../views/open-commit-dialog';
+
+const DIALOG_COMPONENTS = {
+  null: NullDialog,
+  init: InitDialog,
+  clone: CloneDialog,
+  credential: CredentialDialog,
+  issueish: OpenIssueishDialog,
+  commit: OpenCommitDialog,
+};
+
+export default class DialogsController extends React.Component {
+  static propTypes = {
+    // Model
+    request: PropTypes.shape({
+      identifier: PropTypes.string.isRequired,
+      isProgressing: PropTypes.bool.isRequired,
+    }).isRequired,
+
+    // Atom environment
+    workspace: PropTypes.object.isRequired,
+    commands: PropTypes.object.isRequired,
+    config: PropTypes.object.isRequired,
+  };
+
+  state = {
+    requestInProgress: null,
+    requestError: [null, null],
+  }
+
+  render() {
+    const DialogComponent = DIALOG_COMPONENTS[this.props.request.identifier];
+    return <DialogComponent {...this.getCommonProps()} />;
+  }
+
+  getCommonProps() {
+    const {request} = this.props;
+    const accept = request.isProgressing
+      ? async (...args) => {
+        this.setState({requestError: [null, null], requestInProgress: request});
+        try {
+          const result = await request.accept(...args);
+          this.setState({requestInProgress: null});
+          return result;
+        } catch (error) {
+          this.setState({requestError: [request, error], requestInProgress: null});
+          return undefined;
+        }
+      } : (...args) => {
+        this.setState({requestError: [null, null]});
+        try {
+          return request.accept(...args);
+        } catch (error) {
+          this.setState({requestError: [request, error]});
+          return undefined;
+        }
+      };
+    const wrapped = wrapDialogRequest(request, {accept});
+
+    return {
+      config: this.props.config,
+      commands: this.props.commands,
+      workspace: this.props.workspace,
+      inProgress: this.state.requestInProgress === request,
+      error: this.state.requestError[0] === request ? this.state.requestError[1] : null,
+      request: wrapped,
+    };
+  }
+}
+
+function NullDialog() {
+  return null;
+}
+
+class DialogRequest {
```

Comment:
This class is in here so that I can restrict the scope of "things that need to know the List Of All Possible Dialogs" to only this file. Hopefully that'll make it less error-prone to add new ones in the future.